### PR TITLE
Homework 3 and 4 

### PR DESCRIPTION
**Pull request by Maksy-Loop**

### CODE REVIEW

File: Maksym_Moroz/HM_03/homework_3_1.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    """ This function  subtracts one list from another and returns the result. It remove all values (all of its occurrences) from list1, which are present in list2
+        :param list1: It list from which second list is subtracted
+        :param list2: It second list which is subtracted from the first list
+        :type list1: list
+        :type list2: list
+        :return: A new list with deleted values (all its occurrences) from list1, which are present in list2.
+        :rtype: list
+    """
+    counter = 0
+    while True:
```
Comment (medviediev) [inline Maksym_Moroz/HM_03/homework_3_1.py]: It would be better to use for loop here

File: Maksym_Moroz/HM_03/homework_3_2.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    """ This function add number 4 to the argument using concatenation.
+        :param x: It is argument to which the number 4 is added
+        :type x: str
+        :return: Return a string with the added number 4 using concatenation
+        :rtype: str
+    """
+    x = x + "4"
+    return x
+
+def map1(function, list):
```
Comment (medviediev) [inline Maksym_Moroz/HM_03/homework_3_2.py]: arg name should not be list, as list is a builtin data type

File: Maksym_Moroz/HM_03/homework_3_3.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    """ This function filters the list by the value of another function
+        :param function: This function sets the value by which the second argument of the function is filtered
+        :param list: It's list which is filtered
+        :type function: int
+        :type list: list
+        :return: filtered list
+        :rtype: list
+    """
+
+    new_list = [i for i in list if function(i).is_integer()]
```
Comment (medviediev) [inline Maksym_Moroz/HM_03/homework_3_3.py]: What is is_integer? Where did you get those function?

File: Maksym_Moroz/HM_04/homework_4.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+from  collections import OrderedDict
+
+def calculate_words(words):
+    """ This function make сalculate words quantity
+        :param words: list with all the words
+        :type words: list
+        :return: Return word count
+        :rtype: int
+    """
```
Comment (medviediev) [inline Maksym_Moroz/HM_04/homework_4.py]: No need for such a function as simple len() could be used

File: Maksym_Moroz/HM_04/homework_4.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    return sum_words
+
+def unique_words(words):
+    """ This function make extracts unique words
+        :param words: list with all the words
+        :type words: list
+        :return: Return list with unique words
+        :rtype: list
+    """
+    unique_words_dict = set(words)
```
Comment (medviediev) [inline Maksym_Moroz/HM_04/homework_4.py]: No need for such a function as simple set() could be used

File: Maksym_Moroz/HM_04/homework_4.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    print("")
+    print("TOP3: words - frequency (%)")
+    for key, value in dict3.items(): #loop that prints top3 value
+        print(key, value)
+        if counter2 == 2:
+            break
+
+        counter2 += 1
+
+    return True
```
Comment (medviediev) [inline Maksym_Moroz/HM_04/homework_4.py]: Too many redundant stuff in this function

File: Maksym_Moroz/HM_04/homework_4.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                dict3 = OrderedDict(dict2)
+        counter -= 1
+
+    print("Words - frequency (%)")
+
+    for key, value in dict3.items(): #loop that prints key and value
+        print(key, value)
+
+    print("")
+    print("TOP3: words - frequency (%)")
```
Comment (medviediev) [inline Maksym_Moroz/HM_04/homework_4.py]: Function should not print anything, it should return

File: Maksym_Moroz/HM_03/homework_3_3.py (hunk 3 above)
Comment (Maksy-Loop) [reply Maksym_Moroz/HM_03/homework_3_3.py]: Here a function was thought which would filter the list by square root and display only those elements in which the square root is int. Now I understand that this is a little not what you need, this function is only suitable for a special case, I will fix it at my leisure

File: Maksym_Moroz/HM_04/homework_4.py (hunk 4 above)
Comment (Maksy-Loop) [reply Maksym_Moroz/HM_04/homework_4.py]: in the task was to write 4 functions for each sub-item)

File: Maksym_Moroz/HM_04/homework_4.py (hunk 5 above)
Comment (Maksy-Loop) [reply Maksym_Moroz/HM_04/homework_4.py]: in the task was to write 4 functions for each sub-item)

File: Maksym_Moroz/HM_04/homework_4.py (hunk 7 above)
Comment (Maksy-Loop) [reply Maksym_Moroz/HM_04/homework_4.py]: ок

**Pull request merged**